app/getGitSha.gradle

```diff
@@ -0,0 +1,27 @@
+import org.gradle.api.provider.ValueSourceParameters
+import javax.inject.Inject
+
+// Must wrap this in a ValueSource in order to get well-defined fail behavior without confusing Gradle on repeat builds.
+abstract class GitShaValueSource implements ValueSource<String, ValueSourceParameters.None> {
+    @Inject abstract ExecOperations getExecOperations()
+
+    @Override String obtain() {
+        try {
+            def output = new ByteArrayOutputStream()
+
+            execOperations.exec {
+                it.commandLine 'git', 'rev-parse', '--short=8', 'HEAD'
```

Comment:
It should be `--short=7` here.

Reply:
Addressing the fix to #3388.

Reply:
Okay, but why?

Reply:
Default value is 7, which is showing on github web.